src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -227,11 +228,9 @@ private int RunTests(TestPackage package, TestFilter filter)
                 RestoreErrorOutput();
             }
 
-            var writer = new ColorConsoleWriter(!_options.NoColor);
```

Comment:
Does anyone know why we recreated the ConsoleWriter here? This meant my mock wasn't being used - and the two halves of console output were coming through two different writers.

Reply:
The reason is that output from the test itself, which is handled by the first writer, may be redirected. This code goes back to when only a few people worked on NUnit but should have been explained in a comment anyway.

Reply:
Thanks. I wonder if it's possible to re-redirect the original ConsoleWriter instead? That would allow other writers to be passed in to the class. It feels a little odd that the class takes a Writer in the constructor, but then doesn't use that writer to write all the output. 

I'll have another look.

Reply:
I can't see it all very well browsing on the phone but I do remember that adding color made it more complicated and o had to work to ensure that two writers still made one smooth output flow. It may have been changed as well since then.

There are two ways to redirect, --output parameter, which affects only test output and not the final report, and command-line she'll redirection, which affects anything that might go to the console. Taken together, that's four possibilities.